internal/services/containers/kubernetes_cluster_addons_resource_test.go

```diff
@@ -1144,3 +1159,269 @@ resource "azurerm_kubernetes_cluster" "test" {
+    x509_certificate_properties {
+      extended_key_usage = ["1.3.6.1.5.5.7.3.1"]
+      key_usage = [
+        "cRLSign",
+        "dataEncipherment",
+        "digitalSignature",
+        "keyAgreement",
+        "keyCertSign",
+        "keyEncipherment",
+      ]
+      subject_alternative_names {
+        dns_names = ["api.pluginsdk.io"]
+      }
+      subject            = "CN=api.pluginsdk.io"
+      validity_in_months = 1
+    }
+  }
+
+  depends_on = [azurerm_key_vault_access_policy.test]
+}
+
+resource "azurerm_key_vault_key" "test" {
+  name         = "testkeyvaultkey%[1]s"
+  key_vault_id = azurerm_key_vault.test.id
+
+  key_type = "RSA"
+  key_size = 2048
+
+  key_opts = [
+    "decrypt",
+    "encrypt",
+    "sign",
+    "unwrapKey",
+    "verify",
+    "wrapKey",
+  ]
+
+  depends_on = [azurerm_key_vault_access_policy.test]
+}
+
+resource "azurerm_user_assigned_identity" "test" {
+  name                = "acctestUAI-%[1]s"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+}
+
+resource "azurerm_kubernetes_cluster" "test" {
+  name                = "acctestaks%[1]s"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  dns_prefix          = "acctestaks%[1]s"
+
+  linux_profile {
+    admin_username = "acctestuser%[1]s"
+
+    ssh_key {
+      key_data = "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQCqaZoyiz1qbdOQ8xEf6uEu1cCwYowo5FHtsBhqLoDnnp7KUTEBN+L2NxRIfQ781rxV6Iq5jSav6b2Q8z5KiseOlvKA/RF2wqU0UPYqQviQhLmW6THTpmrv/YkUCuzxDpsH7DUDhZcwySLKVVe0Qm3+5N2Ta6UYH3lsDf9R9wTP2K/+vAnflKebuypNlmocIvakFWoZda18FOmsOoIVXQ8HWFNCuw9ZCunMSN62QGamCe3dL5cXlkgHYv7ekJE15IA9aOJcM7e90oeTqo+7HTcWfdu0qQqPWY5ujyMw/llas8tsXY85LFqRnr3gJ02bAscjc477+X+j/gkpFoN1QEmt terraform@demo.tld"
+    }
+  }
+
+  default_node_pool {
+    name           = "default"
+    node_count     = 2
+    vm_size        = "Standard_DS2_v2"
+    vnet_subnet_id = azurerm_subnet.test.id
+    upgrade_settings {
+      max_surge = "10%%"
+    }
+  }
+
+  identity {
+    type = "UserAssigned"
+    identity_ids = [
+      azurerm_user_assigned_identity.test.id,
+    ]
+  }
+
+  network_profile {
+    network_plugin = "kubenet"
+    dns_service_ip = "10.10.0.10"
+    service_cidr   = "10.10.0.0/16"
+  }
+
+  key_vault_secrets_provider {
+    secret_rotation_enabled = true
+  }
+
+  service_mesh_profile {
+    mode                             = "Istio"
+    internal_ingress_gateway_enabled = true
+    external_ingress_gateway_enabled = true
+    certificate_authority {
+      plugin {
+        key_vault_id           = azurerm_key_vault.test.id
+        root_cert_object_name  = azurerm_key_vault_certificate.test_cert1.name
+        cert_chain_object_name = azurerm_key_vault_certificate.test_cert2.name
+        cert_object_name       = azurerm_key_vault_certificate.test_cert3.name
+        key_object_name        = azurerm_key_vault_key.test.name
+      }
+    }
```

Comment:
It looks like there's an unnecessary level of nesting here, since `plugin` can also contain at most 1 element could the schema be flattened and simplified to 
```suggestion
    certificate_authority {
      key_vault_id           = azurerm_key_vault.test.id
      root_cert_object_name  = azurerm_key_vault_certificate.test_cert1.name
      cert_chain_object_name = azurerm_key_vault_certificate.test_cert2.name
      cert_object_name       = azurerm_key_vault_certificate.test_cert3.name
      key_object_name        = azurerm_key_vault_key.test.name
    }
```

Reply:
I added this mainly because:
1.  It reflects the request structure passed to the AKS API.
2. If there are new properties to add under certificate_authority that do not correlate with these plugin settings, it will be easier to manage.

In this case, I believe the benefits of adding an extra layer outweigh the downsides

Reply:
> It reflects the request structure passed to the AKS API.

We don't always map API property names or object structures 1:1 in the provider since the provider has it's own patterns of exposing Azure functionality, and often what's exposed in the API needs simplification for a better user experience in Terraform. In the case of AKS which is a complicated and huge resource, proactively flattening objects also simplifies the code and reduces the cognitive load when working with the resource. It might seem minor for something like this but it adds up over time.

> If there are new properties to add under certificate_authority that do not correlate with these plugin settings, it will be easier to manage.

This can be managed by introducing an additional block under `certificate_authority` for properties that do not correlate to `plugin` settings, or by introducing them as root layer properties at that point in time, _when and if_ that even happens. In this particular case the properties `key_vault_id`, `root_cert_object_name` etc. relate much more to `certificate_authority` than to `plugin`, so further nesting under `plugin` is actually obfuscating than helpful for the user in this case.

Reply:
Thanks, noted.

I have changed the code & doc, ran the tests again and it is passing.
<img width="821" alt="image" src="https://github.com/hashicorp/terraform-provider-azurerm/assets/172344839/f9e113a3-92f2-4406-914c-21c86b5cc5a1">
<img width="831" alt="image" src="https://github.com/hashicorp/terraform-provider-azurerm/assets/172344839/e1440290-2e39-4de8-a0df-4397c6b2b7d3">